crates/bevy_asset/src/io/embedded/mod.rs

```diff
@@ -299,6 +299,73 @@ mod tests {
         assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
     }
 
+    // A blank src_path removes the embedded's file path altogether.
+    #[test]
+    fn embedded_asset_path_from_local_crate_blank_src_path_questionable() {
+        let asset_path = _embedded_asset_path(
+            "my_crate",
+            "".as_ref(),
+            "src/foo/some/deep/path/plugin.rs".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+        assert_eq!(asset_path, Path::new("my_crate/the/asset.png"));
+    }
+
+    #[should_panic(expected = "Failed to find src_prefix \"src\" in \"\"")]
+    #[test]
+    fn embedded_asset_path_from_local_crate_blank_file_path() {
+        let asset_path = _embedded_asset_path(
+            "my_crate",
+            "src".as_ref(),
+            "".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+        assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
+    }
+
+    // Can have extraneous slashses at the end of source_path.
+    #[test]
+    fn embedded_asset_path_from_local_crate_extraneous_slalsh() {
+        let asset_path = _embedded_asset_path(
+            "my_crate",
+            "src/".as_ref(),
+            "src/foo/plugin.rs".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+        assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
+
+        let asset_path = _embedded_asset_path(
+            "my_crate",
+            "src/////".as_ref(),
+            "src/foo/plugin.rs".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+        assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
+    }
+
+    #[test]
+    #[should_panic(expected = "Failed to find src_prefix \"NOT-THERE\" in \"src")]
+    fn embedded_asset_path_from_local_crate_bad_src() {
+        let _asset_path = _embedded_asset_path(
+            "my_crate",
+            "NOT-THERE".as_ref(),
+            "src/foo/plugin.rs".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+    }
+
+    // Trying to provoke a panic from `parent().unwrap()`, but it's solid.
```

Comment:
I'm not sure how I feel about this one. It almost looks like a regression test for a bug that never existed.

Reply:
Agreed. Deleted.